### PR TITLE
Loki Auditing

### PR DESCRIPTION
**Pull request by pandabear15**
In addition to logging to a QLDB, logging to Loki, a highly available logging service, makes it simple to retrieve audit logs in the future, at the cost of the lack of assurance of modification, security, etc. Adding loki auditing is not intended to replace the QLDB, indeed, the QLDB is intended to be the source of truth, and Loki logs are intended to solely be an accessible way to retrieve the same logs. If auditors need the cryptographic proof that the logs are what they should be, then they should access the QLDB instead of Loki.

### CODE REVIEW

File: conf/logback.loki.xml [hunk 1]
```diff
@@ -13,6 +13,20 @@
             <sortByTime>true</sortByTime>
         </format>
     </appender>
+    <appender name="AUDIT" class="com.github.loki4j.logback.Loki4jAppender">
+        <http>
+            <url>http://localhost:3200/loki/api/v1/push</url>
```
Comment (cody-constine-ttd) [inline conf/logback.loki.xml:18]: Can you make a .local config if you are going to commit localhost:3200

File: conf/logback.loki.xml [hunk 2]
```diff
@@ -21,8 +35,12 @@
         </encoder>
     </appender>
 
+    <logger name="com.uid2.admin.audit" additivity="false">
+        <appender-ref ref="AUDIT"/>
+    </logger>
+
     <root level="DEBUG">
-        <appender-ref ref="LOKI" />
+        <appender-ref ref="LOKI"   />
```
Comment (cody-constine-ttd) [inline conf/logback.loki.xml:43]: *nit* extra whitespace

File: conf/logback.loki.xml (hunk 1 above)
Comment (pandabear15) [reply conf/logback.loki.xml:18]: ppppffftt this is supposed to be loki:3200

File: conf/logback.loki.xml (hunk 2 above)
Comment (pandabear15) [reply conf/logback.loki.xml]: fixed

File: pom.xml [hunk 3]
```diff
@@ -89,6 +89,21 @@
             <version>${vertx.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (haitaottd) [inline pom.xml:92]: this is probably needed too:
```
<dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
            <version>1.7.26</version>
</dependency>
```

File: src/main/java/com/uid2/admin/audit/QLDBAuditWriter.java [hunk 4]
```diff
@@ -17,6 +17,7 @@
 public class QLDBAuditWriter implements AuditWriter{
     private static final IonSystem ionSys = IonSystemBuilder.standard().build();
     private static final Logger logger = LoggerFactory.getLogger(QLDBAuditWriter.class);
+    private static final Logger auditLogger = LoggerFactory.getLogger("com.uid2.admin.audit");
```
Comment (haitaottd) [inline src/main/java/com/uid2/admin/audit/QLDBAuditWriter.java:20]: do you need booth loggers?

Comment (pandabear15) [reply src/main/java/com/uid2/admin/audit/QLDBAuditWriter.java:20]: Yes, one logger writes to the separate loki audit log, the other is for error writing / standard loki log

File: pom.xml (hunk 3 above)
Comment (pandabear15) [reply pom.xml:92]: will add

**Pull request merged**